src/Metadata/RootMetadata.php

```diff
@@ -70,19 +78,19 @@ public function getRoles(bool $allowUntrustedAccess = false): array
      * @param boolean $allowUntrustedAccess
      *   Whether this method should access even if the metadata is not trusted.
      *
-     * @return \ArrayObject
-     *   An ArrayObject of keys information where the array keys are the key ids and
-     *   the values are arrays with the following values:
-     *   - keyid_hash_algorithms (string[]): The key id algorithms used.
-     *   - keytype (string): The key type.
-     *   - keyval (string[]): A single item array where key value is 'public'
-     *     and the value is the public key.
-     *   - scheme (string): The key scheme.
+     * @return \Tuf\Key[]
+     *   The keys for the metadata.
      */
-    public function getKeys(bool $allowUntrustedAccess = false): \ArrayObject
+    public function getKeys(bool $allowUntrustedAccess = false): array
     {
-        $this->ensureIsTrusted($allowUntrustedAccess);
-        return $this->getSigned()['keys'];
+        if (!isset($this->keys)) {
+            $this->ensureIsTrusted($allowUntrustedAccess);
+            $this->keys = [];
```

Comment:
Is it really necessary to store `$this->keys`? Feels like unnecessary state mutation in this class, which is probably something to be avoided if possible. That said, I don't feel strongly about it.

Reply:
I added this because `RootMetadata` has no setters so you could never change the keys once the object was created was thinking of performance if `getKeys()` was called more than 1x. But right now it is not and this class is not final(maybe it should be) so we don't know a subclass won't add a setter for keys.

Removing